seed/views/v3/properties.py

```diff
@@ -45,3 +208,769 @@ def labels(self, request):
+
+        for h in history:
+            h['state'] = PropertyStateSerializer(h['state_data']).data
+            del h['state_data']
+            del h['state_id']
+
+        return history, master
+
+    @swagger_auto_schema_org_query_param
+    @api_endpoint_class
+    @ajax_request_class
+    def retrieve(self, request, pk=None):
+        """
+        Get property details
+        """
+        result = self._get_property_view(pk)
+        if result.get('status', None) != 'error':
+            property_view = result.pop('property_view')
+            result = {'status': 'success'}
+            result.update(PropertyViewSerializer(property_view).data)
+            # remove PropertyView id from result
+            result.pop('id')
+
+            # Grab extra_data columns to be shown in the result
+            organization_id = request.query_params['organization_id']
+            all_extra_data_columns = Column.objects.filter(
+                organization_id=organization_id,
+                is_extra_data=True,
+                table_name='PropertyState').values_list('column_name', flat=True)
+
+            result['state'] = PropertyStateSerializer(property_view.state,
+                                                      all_extra_data_columns=all_extra_data_columns).data
+            result['taxlots'] = self._get_taxlots(property_view.pk)
+            result['history'], master = self.get_history(property_view)
+            result = update_result_with_master(result, master)
+            return JsonResponse(result, encoder=PintJSONEncoder, status=status.HTTP_200_OK)
+        else:
+            return JsonResponse(result, status=status.HTTP_404_NOT_FOUND)
+
+    @swagger_auto_schema(
+        manual_parameters=[AutoSchemaHelper.query_org_id_field()],
+        request_body=UpdatePropertyPayloadSerializer,
+    )
+    @api_endpoint_class
+    @ajax_request_class
+    def update(self, request, pk=None):
+        """
+        Update a property and run the updated record through a match and merge
+        round within it's current Cycle.
+
+        - looks up the property view
+        - casts it as a PropertyState
+        - builds a hash with all the same keys as the original property state
+        - checks if any fields have changed
+        - if nothing has changed, return 422 - Really?  Not sure how I feel about that one, it *is* processable
+        - get the property audit log for this property state
+        - if the new property state has extra_data, the original extra_data is update'd
+        - and then whoa stuff about the audit log?
+        - I'm going to assume 'Import Creation' is the key I'm looking for
+        - create a serializer for the new property state
+        - if it's valid, save this new serialized data to the db
+        - assign it to the original property view and save the property view
+        - create a new property audit log for this change
+        - return a 200 if created
+        """
```

Comment:
We should probably revise the roadmap above to exclude mental notes.  Specifically on lines 810, 813, 814.